fix(gatsby) write empty files when no pages created or no pages removed

tomknapen opened the pull request:
## Description

Context: When using the experimental page build on data changes, the output files `newPages.txt` and `deletedPages.txt` are generated when new pages were added, pages were updated or pages were deleted during the build process.

Problem: The output files `newPages.txt` and `deletedPages.txt` are not up to date on subsequent builds, if there are no changes to them. Using these files for anything post-build (e.g. generating a report) might result in wrong feedback. 

Resolution: The files get written with empty contents if they do not change during the new build.

### Documentation

Removed documentation saying the files are not being generated in `docs/docs/conditional-page-builds.md`.

pieh commented:
/cc @dominicfallows @StuartRayson

I second creating empty files over deleting those as well.

Nice catch!

dominicfallows commented:
Agree, great catch. 

I would also agree to generating empty files on no change builds.

tomknapen commented:
Was not sure about doing that, as the docs said "they would not be generated". I have adjusted the PR accordingly: empty files are generated when no changes happen (create/delete), and I removed the docs saying they are not generated when nothing changes.

tomknapen commented:
Edited the PR description and title accordingly.

freiksenet commented:
In this sense this is a "more correct" behaviour in a sense that it's consistent. However I feel that we probably shouldn't have had this new line originally, but it's not worth subtly breaking this for people because of this.

dominicfallows commented:
Yes, I'm discussing internally if there was a reason for the new line, I agree that it might not need to be there

dominicfallows commented:
We have looked back, and the new line resolved issues/equalized a way of parsing the files across different scripts, Node, Bash etc

tomknapen commented:
Would you prefer me to change this behavior, or leave it as is? Should we do it differently for the empty files?

dominicfallows commented:
The problem seems to be when using Bash to read line-by-line, we need a terminating newline. I suppose that means that the empty file doesn't need it, but the populated file does.

freiksenet commented:
Thanks a lot for this fix @tomknapen!
